Packet++/header/Layer.h

```diff
@@ -211,8 +211,8 @@ namespace pcpp
 		      m_PrevLayer(nullptr), m_IsAllocatedInPacket(false)
 		{}
 
-		Layer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
-		    : m_Data(data), m_DataLen(dataLen), m_Packet(packet), m_Protocol(UnknownProtocol), m_NextLayer(nullptr),
+		Layer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet, ProtocolType protocol = UnknownProtocol)
```

Comment:
Why do we need a default value of `UnknownProtocol`? 🤔

Reply:
To keep compatibility as some constructors still use the non-protocol version to initialize to `UnknownProtocol` and set the `m_protocol` manually. Usually in some kind of `init` method that gets executed in the body of the constructor.